src/main.c

```diff
@@ -130,15 +137,26 @@ static void load_port_config(struct uci_context *uci, struct uci_section *s)
 
 static void load_global_config(struct uci_context *uci, struct uci_section *s)
 {
-	const char *budget, *guardband;
+	const char *budget, *guardband, *i2c_bus, *i2c_addr;
 
 	budget = uci_lookup_option_string(uci, s, "budget");
 	guardband = uci_lookup_option_string(uci, s, "guard");
+	i2c_bus = uci_lookup_option_string(uci, s, "i2c_bus");
+	i2c_addr = uci_lookup_option_string(uci, s, "i2c_addr");
 
 	config.budget = budget ? strtof(budget, NULL) : 31.0;
 	config.budget_guard = config.budget / 10;
 	if (guardband)
 		config.budget_guard = strtof(guardband, NULL);
+	if (i2c_bus) {
+		config.i2c_bus = strtol(i2c_bus, NULL, 10);
+		if (i2c_addr)
+			config.i2c_addr = strtol(i2c_addr, NULL, 10);
+		else
+			config.i2c_addr = 0x20; // I2C address on both known devices
```

Comment:
I'll follow you suggestion here. But I don't believe this is going to be board specific, since there is some generic SDK behind that. Anyway, we could also do auto-detection here.